tests/data/ar/Item.php

```diff
@@ -20,4 +20,11 @@ public function getCategory()
     {
         return $this->hasOne(Category::className(), ['id' => 'category_id']);
     }
+
+    //relation with dynamic alias
+    public function getCategoryWithAlias()
```

Comment:
has one relation exists, need write test for exists relations, then all test cover you suggestion